plugin/pkg/admission/resourcequota/admission.go

```diff
@@ -190,52 +190,66 @@ func IncrementUsage(a admission.Attributes, status *api.ResourceQuotaStatus, cli
 			}
 		}
 	}
-	// handle memory/cpu constraints, and any diff of usage based on memory/cpu on updates
-	if a.GetResource() == "pods" && (set[api.ResourceMemory] || set[api.ResourceCPU]) {
-		pod := obj.(*api.Pod)
-		deltaCPU := resourcequotacontroller.PodCPU(pod)
-		deltaMemory := resourcequotacontroller.PodMemory(pod)
-		// if this is an update, we need to find the delta cpu/memory usage from previous state
-		if a.GetOperation() == admission.Update {
-			oldPod, err := client.Pods(a.GetNamespace()).Get(pod.Name)
-			if err != nil {
-				return false, err
+
+	if a.GetResource() == "pods" {
+		for _, resourceName := range []api.ResourceName{api.ResourceMemory, api.ResourceCPU} {
+			// ignore tracking the resource if its not in the quota document
+			if !set[resourceName] {
+				continue
 			}
-			oldCPU := resourcequotacontroller.PodCPU(oldPod)
-			oldMemory := resourcequotacontroller.PodMemory(oldPod)
-			deltaCPU = resource.NewMilliQuantity(deltaCPU.MilliValue()-oldCPU.MilliValue(), resource.DecimalSI)
-			deltaMemory = resource.NewQuantity(deltaMemory.Value()-oldMemory.Value(), resource.DecimalSI)
-		}
 
-		hardMem, hardMemFound := status.Hard[api.ResourceMemory]
-		if hardMemFound {
-			if set[api.ResourceMemory] && resourcequotacontroller.IsPodMemoryUnbounded(pod) {
-				return false, fmt.Errorf("Limited to %s memory, but pod has no specified memory limit", hardMem.String())
+			// if we do not yet know how much of the current resource is used, we cannot accept any request
+			hard, hardFound := status.Hard[resourceName]
+			if !hardFound {
+				continue
 			}
-			used, usedFound := status.Used[api.ResourceMemory]
+
+			used, usedFound := status.Used[resourceName]
 			if !usedFound {
-				return false, fmt.Errorf("Quota usage stats are not yet known, unable to admit resource until an accurate count is completed.")
+				return false, fmt.Errorf("Unable to admit pod until quota usage stats are calculated.")
 			}
-			if used.Value()+deltaMemory.Value() > hardMem.Value() {
-				return false, fmt.Errorf("Limited to %s memory", hardMem.String())
-			} else {
-				status.Used[api.ResourceMemory] = *resource.NewQuantity(used.Value()+deltaMemory.Value(), resource.DecimalSI)
-				dirty = true
+
+			// the amount of resource being requested, or an error if it does not make a request that is tracked
+			pod := obj.(*api.Pod)
+			delta, err := resourcequotacontroller.PodRequests(pod, resourceName)
```

Comment:
Where do you see `delta` used outside this scope? I assume you're referring to L239, but that's in the same scope.

(FWIW I found it impossible to review this file using side-by-side mode and had to click "View" to view the actual file)